pkg/model/hornet/transaction_metadata.go

```diff
@@ -124,6 +127,23 @@ func (m *TransactionMetadata) SetConflicting(conflicting bool) {
 	}
 }
 
+func (m *TransactionMetadata) SetRootSnapshotIndexes(yrtsi milestone.Index, ortsi milestone.Index, rtsci milestone.Index) {
+	m.Lock()
+	defer m.Unlock()
+
+	m.youngestRootSnapshotIndex = yrtsi
+	m.oldestRootSnapshotIndex = ortsi
+	m.rootSnapshotCalculationIndex = rtsci
+	m.SetModified(true)
+}
+
+func (m *TransactionMetadata) GetRootSnapshotIndexes() (milestone.Index, milestone.Index, milestone.Index) {
```

Comment:
maybe give the return values a label?
`(yrtsi milestone.Index, ortsi milestone.Index, rtsci milestone.Index)`